pkg/collector/scheduler.go

```diff
@@ -0,0 +1,208 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package collector
+
+import (
+	"expvar"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	"github.com/DataDog/datadog-agent/pkg/collector/loaders"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var (
+	schedulerErrs *expvar.Map
+	errorStats    = newCollectorErrors()
+	// Scheduler is the global check scheduler instance
+	Scheduler *CheckScheduler
+)
+
+func init() {
+	schedulerErrs = expvar.NewMap("CheckScheduler")
+	schedulerErrs.Set("LoaderErrors", expvar.Func(func() interface{} {
+		return errorStats.getLoaderErrors()
+	}))
+	schedulerErrs.Set("RunErrors", expvar.Func(func() interface{} {
+		return errorStats.getRunErrors()
+	}))
+}
+
+// CheckScheduler is the check scheduler
+type CheckScheduler struct {
+	configToChecks map[string][]check.ID // cache the ID of checks we load for each config
+	loaders        []check.Loader
+	collector      *Collector
+	m              sync.RWMutex
+}
+
+// InitCheckScheduler creates and returns the global check scheduler
+func InitCheckScheduler(collector *Collector) *CheckScheduler {
```

Comment:
I think we could renamed this method to GetCheckScheduler and use `lazy loading`.